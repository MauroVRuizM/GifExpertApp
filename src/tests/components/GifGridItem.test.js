import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Prueba';
    const url = 'https://prueba.png';
    
    let wrapper = shallow(<GifGridItem title={ title } url={ url } />); /** Por el Intellicense */

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={ title } url={ url } />);
    });

    test('Debe mostrar <GifGridItem/> correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un pÃ¡rrafo con el title ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    
    test('Debe tener la imagen igual al url y alt de los props ', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    
    test('Debe tener animate__flash ', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__flash')).toBe(true);
    });
    
})
