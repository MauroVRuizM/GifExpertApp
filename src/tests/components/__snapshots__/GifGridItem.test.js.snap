// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar <GifGridItem/> correctamente  1`] = `
<div
  className="card animate__animated animate__flash"
>
  <img
    alt="Prueba"
    src="https://prueba.png"
  />
  <p>
    Prueba
  </p>
</div>
`;
